drivers/interrupt_controller/intc_exti_stm32.c

```diff
@@ -0,0 +1,372 @@
+	defined(CONFIG_SOC_SERIES_STM32F7X) || \
+	defined(CONFIG_SOC_SERIES_STM32H7X) || \
+	defined(CONFIG_SOC_SERIES_STM32H7RSX) || \
+	defined(CONFIG_SOC_SERIES_STM32L1X) || \
+	defined(CONFIG_SOC_SERIES_STM32L4X) || \
+	defined(CONFIG_SOC_SERIES_STM32G4X)
+	const struct device *const clk = DEVICE_DT_GET(STM32_CLOCK_CONTROL_NODE);
+
+	if (!device_is_ready(clk)) {
+		LOG_ERR("Clock control device not ready");
+		return -ENODEV;
+	}
+
+	struct stm32_pclken pclken = {
+#if defined(CONFIG_SOC_SERIES_STM32H7X)
+		.bus = STM32_CLOCK_BUS_APB4,
+		.enr = LL_APB4_GRP1_PERIPH_SYSCFG
+#elif defined(CONFIG_SOC_SERIES_STM32H7RSX)
+		.bus = STM32_CLOCK_BUS_APB4,
+		.enr = LL_APB4_GRP1_PERIPH_SBS
+#else
+		.bus = STM32_CLOCK_BUS_APB2,
+		.enr = LL_APB2_GRP1_PERIPH_SYSCFG
+#endif /* CONFIG_SOC_SERIES_STM32H7X */
+	};
+
+	ret = clock_control_on(clk, (clock_control_subsys_t) &pclken);
+#endif
+	return ret;
+}
+
+/**
+ * @brief Initializes the EXTI GPIO interrupt controller driver
+ */
+static int stm32_exti_init(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	return stm32_exti_enable_registers();
+}
+
+/**
+ * @brief Enable EXTI interrupts.
+ *
+ * @param line_num	EXTI line number
+ * @param line	LL EXTI line
+ */
+static void stm32_exti_enable_it(uint32_t line_num, uint32_t line)
+{
+	EXTI_FN_HANDLER(EXTI_ENABLE_IT, line_num, line);
+}
+
+/**
+ * @brief Disable EXTI interrupts.
+ *
+ * @param line_num	EXTI line number
+ * @param line	LL EXTI line
+ */
+static void stm32_exti_disable_it(uint32_t line_num, uint32_t line)
+{
+	EXTI_FN_HANDLER(EXTI_DISABLE_IT, line_num, line);
+}
+
+/**
+ * @brief Enables rising trigger for specified EXTI line
+ *
+ * @param line_num	EXTI line number
+ * @param line	LL EXTI line
+ */
+static void stm32_exti_enable_rising_trig(uint32_t line_num, uint32_t line)
+{
+	EXTI_FN_HANDLER(EXTI_ENABLE_RISING_TRIG, line_num, line);
+}
+
+/**
+ * @brief Disables rising trigger for specified EXTI line
+ *
+ * @param line_num	EXTI line number
+ * @param line	LL EXTI line
+ */
+static void stm32_exti_disable_rising_trig(uint32_t line_num, uint32_t line)
+{
+	EXTI_FN_HANDLER(EXTI_DISABLE_RISING_TRIG, line_num, line);
+}
+
+/**
+ * @brief Enables falling trigger for specified EXTI line
+ *
+ * @param line_num	EXTI line number
+ * @param line	LL EXTI line
+ */
+static void  stm32_exti_enable_falling_trig(uint32_t line_num, uint32_t line)
+{
+	EXTI_FN_HANDLER(EXTI_ENABLE_FALLING_TRIG, line_num, line);
+}
+
+/**
+ * @brief Disables falling trigger for specified EXTI line
+ *
+ * @param line_num	EXTI line number
+ * @param line	LL EXTI line
+ */
+static void stm32_exti_disable_falling_trig(uint32_t line_num, uint32_t line)
+{
+	EXTI_FN_HANDLER(EXTI_DISABLE_FALLING_TRIG, line_num, line);
+}
+
+/**
+ * @brief Selects EXTI trigger mode
+ *
+ * @param line_num	EXTI line number
+ * @param line	LL EXTI line
+ * @param mode	EXTI mode
+ */
+static void stm32_exti_select_line_trigger(uint32_t line_num, uint32_t line,
+										   uint32_t trg)
```

Comment:
Nit: indentation

Reply:
See my comment about GitHub's representation. I'll do a screenshot later.
In such unclear cases I would prefer to rely on the checkpatch.pl as a single source of truths. Since it does not complain, let's leave it. Please refer to the corresponding stage output and you will see, that there is even no warnings.

Reply:
This is strange as we usually manage to get same file representation in GH and editors (VSCode, atom for me).
Are you using project recommendation about 8 char long tabs ?

Not blocking anyway.

Reply:
Thanks but don't bother. Really not blocking.